pkgdb/src/resolver/mixins.cc

```diff
@@ -303,7 +303,7 @@ argparse::Argument &
 EnvironmentMixin::addLockfileOption( argparse::ArgumentParser & parser )
 {
   return parser.add_argument( "--lockfile" )
-    .help( "the path to the projects existing `manifest.lock' file." )
+    .help( "the path to a lockfile, either the project's `manifest.lock` or the global lock" )
```

Comment:
I'm thinking there can only be one lockfile. And `flox` should decide which to use rather than `pkgdb`. Wanted to flag as I'm not sure we've really discussed this behavior. And search and show will have to be updated to use this lockfile.

Reply:
For cases where the global manifest has inputs that don't exist in the project we'll want two; but for now that approach for a single lock sounds good to me 👍

Reply:
I thought in that case we were copying the input into the environment, so that if it gets shared, whoever pulls the environment can still build and update it?